74.md

```diff
@@ -0,0 +1,102 @@
+# NIP-74
+
+## Enhanced Sorting and Filtering for Social Media Posts
+
+`draft` `optional`
+
+## Abstract
+
+This NIP defines `kind:33031`: a standardized JSON configuration for sorting and filtering social media posts. The configuration allows users to customize how posts are displayed based on various parameters and metadata.
+
+## Motivation
+
+Various NOSTR client frameworks need efficient mechanisms to display relevant posts to users. Sorting and filtering based on user preferences, engagement metrics, and content metadata can enhance the user experience by presenting the most pertinent information. This proposal aims to establish a standard format for configuring such sorting and filtering mechanisms, allowing for "portability" of a users algorithmic preferences across existing clients and any future implementations.
+
+## Specification
+
+The configuration is defined using a JSON object with the following structure:
+
+```json
+{
+  "kind": 33031,
+  "created_at": 1716994345,
+  "pubkey": "...",
+  "id": "...",
+  "sortingAlgorithmConfig": {
```

Comment:
`.sortingAlgorithmConfig` should be the `.content` and the value should be JSON-serialized string. If not, I think an event will not be signed correctly.

Reply:
I got you...